### PR TITLE
docker: Multiple fixes

### PR DESCRIPTION
**Pull request by MattKobayashi**
## Change Summary

- Update Debian base image to 12
- Add `--break-system-packages` option to `pip3 install` command
- Remove `chmod a+s` from Dockerfile (newer versions of `gosu` in particular don't support this)
- Add `sudo` to relevant commands in `entrypoint.sh`
- Add `UID_MIN` key to `adduser` command (allows the container to run on MacOS)
- Formatting and spelling fixes

## Backport

- sagitta
- equuleus

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-documentation/blob/current/CONTRIBUTING.md) document

### CODE REVIEW

**Review by rebortg — CHANGES_REQUESTED** [review]
Did you test the build and usage? I Build this on Mac but got an error when i want to build the docu:
```
docker run --rm -it -v "$(pwd)":/vyos -w /vyos/docs \
  -e GOSU_UID=$(id -u) -e GOSU_GID=$(id -g) vyos-docsv2 make html
Current UID/GID: 501/20
sudo: exec: command not found
```

**Comment by MattKobayashi** [comment]
@rebortg I did test it at the time, but for some reason it isn't working now. I've added another commit to fix the issue, tested working on macOS 14.4.1.

**Comment by c-po** [comment]
@MattKobayashi can you please add a comment why `--key UID_MIN=500` the min UID needs to be 500? Maybe a MacOS thing?

**Comment by MattKobayashi** [comment]
@c-po That's correct, macOS UIDs begin at 501. This is actually part of the [Linux Standard Base specification](http://refspecs.linux-foundation.org/LSB_5.0.0/LSB-Core-generic/LSB-Core-generic/uidrange.html).

File: docker/entrypoint.sh [hunk 1]
```diff
@@ -20,13 +20,13 @@ echo "Current UID/GID: $NEW_UID/$NEW_GID"
 
 # Create UNIX group on the fly if it does not exist
 if ! grep -q $NEW_GID /etc/group; then
-    groupadd --gid $NEW_GID $USER_NAME
+    sudo groupadd --gid $NEW_GID $USER_NAME
```
Comment (c-po) [inline docker/entrypoint.sh]: Why is `sudo` required? I remember the container inside runs as root while building it?

Comment (MattKobayashi) [reply docker/entrypoint.sh]: Agreed, these sudos aren't necessary. I've pushed a commit that removes them.

**Comment by c-po** [comment]
@MergifyIo backport sagitta

**Pull request merged**